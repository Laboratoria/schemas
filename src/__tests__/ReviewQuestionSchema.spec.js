const mongoose = require('mongoose/browser');
const { ReviewQuestionSchema } = require('../../')(mongoose);

describe('ReviewQuestionSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ReviewQuestionSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should fail validation when multiple-choice question is missing options', (done) => {
    const doc = new mongoose.Document({
      id: 'foo',
      text: 'bar',
      type: 'multiple-choice',
      visibility: 'private',
    }, ReviewQuestionSchema);

    doc.validate((err) => {
      expect(err.message).toBe('Options is required when type is multiple-choice');
      done();
    });
  });

  it('should fail validation when question is missing id', () => {
    const doc = new mongoose.Document({
      text: 'Hola',
      type: 'multiple-choice',
      options: ['1', '2'],
      visibility: 'private',
    }, ReviewQuestionSchema);

    return doc.validate()
      .catch(err => expect(err.message).toMatchSnapshot());
  });

  it('should fail validation when wrong type', () => {
    const doc = new mongoose.Document({
      id: 'bar',
      text: 'Hola',
      type: 'other',
      visibility: 'public',
    }, ReviewQuestionSchema);

    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should fail validation when question options not a Array', () => {
    const doc = new mongoose.Document({
      id: 'bar',
      text: 'foo',
      type: 'multiple-choice',
      options: 3,
    }, ReviewQuestionSchema);

    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should validate good review Open Question ', () => {
    const doc = new mongoose.Document({
      id: 'foo',
      text: 'foo',
      type: 'open',
      visibility: 'public',
    }, ReviewQuestionSchema);

    return doc.validate();
  });

  it('should validate good review Multiple-choice Question', () => {
    const doc = new mongoose.Document({
      id: 'bar',
      text: 'foo',
      type: 'multiple-choice',
      visibility: 'private',
      options: ['1', '2'],
    }, ReviewQuestionSchema);

    return doc.validate();
  });
});
