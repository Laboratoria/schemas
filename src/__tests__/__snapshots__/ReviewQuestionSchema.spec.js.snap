// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewQuestionSchema should fail validation when fields missing 1`] = `
Object {
  "id": [ValidatorError: Path \`id\` is required.],
  "text": [ValidatorError: Path \`text\` is required.],
  "type": [ValidatorError: Path \`type\` is required.],
  "visibility": [ValidatorError: Path \`visibility\` is required.],
}
`;

exports[`ReviewQuestionSchema should fail validation when question is missing id 1`] = `"Validation failed: id: Path \`id\` is required."`;

exports[`ReviewQuestionSchema should fail validation when question options not a Array 1`] = `
Object {
  "visibility": [ValidatorError: Path \`visibility\` is required.],
}
`;

exports[`ReviewQuestionSchema should fail validation when wrong type 1`] = `
Object {
  "type": [ValidatorError: \`other\` is not a valid enum value for path \`type\`.],
}
`;
